### PR TITLE
Requirement definition fixes

### PR DESCRIPTION
**Pull request by anzoman**
Resolve bug that was found in TOSCA parser for service template and add more meaningful error messages for relationship requirements.

### CODE REVIEW

File: src/opera/parser/tosca/v_1_3/requirement_definition.py [hunk 1]
```diff
@@ -22,3 +22,9 @@ def normalize(cls, yaml_node):
         if isinstance(yaml_node.value, str):
             return Node({Node("capability"): yaml_node})
         return yaml_node
+
+    @classmethod
+    def validate(cls, yaml_node):
+        super().validate(yaml_node)
+        if "relationship" in yaml_node.bare and not isinstance(yaml_node.bare["relationship"], str):
+            cls.abort("Expected a relationship string reference.", yaml_node.loc)
```
Comment (tadeboro) [inline src/opera/parser/tosca/v_1_3/requirement_definition.py:30]: Maybe we can be more specific here: "Expected a relationship type name as a 'relationship' value."

Comment (anzoman) [reply src/opera/parser/tosca/v_1_3/requirement_definition.py]: Ok, I chaged that as you suggested.

**Comment by anzoman** [comment]
I also added the relationship validation to the requirement assignment so there will be no need to do this in a separate PR.

File: src/opera/parser/tosca/v_1_3/requirement_assignment.py [hunk 2]
```diff
@@ -25,3 +25,9 @@ def normalize(cls, yaml_node):
         if isinstance(yaml_node.value, str):
             return Node({Node("node"): yaml_node})
         return yaml_node
+
+    @classmethod
+    def validate(cls, yaml_node):
+        super().validate(yaml_node)
+        if "relationship" in yaml_node.bare and not isinstance(yaml_node.bare["relationship"], str):
+            cls.abort("Expected a relationship type name as a 'relationship' value.", yaml_node.loc)
```
Comment (tadeboro) [inline src/opera/parser/tosca/v_1_3/requirement_assignment.py:33]: Here, we are expecting a relationship template name, not type.

Comment (anzoman) [reply src/opera/parser/tosca/v_1_3/requirement_assignment.py]: Expecting type was changed to expecting template.

**Pull request merged**